gateway/src/http_entrypoints.rs

```diff
@@ -0,0 +1,357 @@
+    _path: Path<PathSpComponent>,
+    // TODO do we need a timeout?
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    todo!()
+}
+
+/// Power off an SP component
+///
+/// Components whose power state cannot be changed will always return an error.
+#[endpoint {
+    method = POST,
+    path = "/sp/{type}/{slot}/component/{component}/power_off",
+}]
+async fn sp_component_power_off(
+    _rqctx: Arc<RequestContext<GatewayService>>,
+    _path: Path<PathSpComponent>,
+    // TODO do we need a timeout?
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    todo!()
+}
+
+/// List SPs via Ignition
+///
+/// List the SPs via the Ignition controller. This mechanism retrieves less
+/// information than over the management network, however it is lower latency
+/// and has fewer moving pieces that could result in delayed responses or
+/// unknown states.
+///
+/// This interface queries ignition via its associated SP. As this interface
+/// may be unreliable, consumers may optionally override the default.
+#[endpoint {
+    method = GET,
+    path = "/ignition",
+}]
+async fn ignition_list(
+    _rqctx: Arc<RequestContext<GatewayService>>,
+    _query: Query<PaginationParams<EmptyScanParams, SpIdentifier>>,
+) -> Result<HttpResponseOk<ResultsPage<SpIgnitionInfo>>, HttpError> {
+    todo!()
+}
+
+/// Get SP info via Ignition
+///
+/// Retreive information for an SP via the Ignition controller. This is lower
+/// latency and has fewer possible failure modes than querying the SP over the
+/// management network.
+#[endpoint {
+    method = GET,
+    path = "/ignition/{type}/{slot}",
+}]
+async fn ignition_get(
+    _rqctx: Arc<RequestContext<GatewayService>>,
+    _path: Path<PathSp>,
+) -> Result<HttpResponseOk<SpIgnitionInfo>, HttpError> {
+    todo!()
+}
+
+/// Power on an SP via Ignition
+#[endpoint {
+    method = POST,
+    path = "/sp/{type}/{slot}/power_on",
+}]
+async fn ignition_power_on(
+    _rqctx: Arc<RequestContext<GatewayService>>,
+    _path: Path<PathSp>,
+) -> Result<HttpResponseOk<SpIgnitionInfo>, HttpError> {
+    todo!()
+}
+
+/// Power off an SP via Ignition
+#[endpoint {
+    method = POST,
+    path = "/sp/{type}/{slot}/power_off",
+}]
+async fn ignition_power_off(
+    _rqctx: Arc<RequestContext<GatewayService>>,
+    _path: Path<PathSp>,
+) -> Result<HttpResponseOk<SpIgnitionInfo>, HttpError> {
+    todo!()
+}
+
+// TODO
+// The gateway service will get asynchronous notifications both from directly
+// SPs over the management network and indirectly from Ignition via the Sidecar
+// SP.
+// TODO The Ignition controller will send an interrupt to its local SP. Will
+// that SP then notify both gateway services or just its local gateway service?
+// Both Ignition controller should both do the same thing at about the same
+// time so is there a real benefit to them both sending messages to both
+// gateways? This would cause a single message to effectively be replicated 4x
+// (Nexus would need to dedup these).
```

Comment:
Presumably, this would be for relatively uncommon events, no? I think the increased traffic to Nexus probably isn't a *huge* downside relative to the availability gain, as long as Nexus can reliably do the de-duplication, even with some time-skew between event arrival.

Reply:
Perhaps uncommon but expected. Yes, the work to dedup isn't significant.

This TODO was intended to discuss that API endpoint we'll need to add to Nexus of which MGS will be consumer. I guess it's kind of like oximeter in that regard.